internal/wasm/store_module_list.go

```diff
@@ -100,7 +107,25 @@ func (s *Store) requireModuleName(moduleName string) error {
 	}
 	s.moduleList = node
 	s.nameToNode[moduleName] = node
-	return nil
+	return node, nil
+}
+
+func (s *Store) registerAnonymous() *moduleListNode {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+
+	// add the newest node to the moduleNamesList as the head.
+	node := &moduleListNode{
+		name: "",
+		next: s.moduleList,
+	}
```

Comment:
We could move this instantiation before acquiring the mutex to keep the critical section to a minimum.

Reply:
good call! Thats an optimization I can add to `requireModuleName` also